fix: update handler to not delete banner

sean-brydon opened the pull request:
## What does this PR do?
Solves the disappearing org banner that has been happning for some time 

## How to test
Upload a banner -> save
Edit bio -> save (verify banner still there)
Remove banner -> save (banner gone)

sean-brydon committed:
Fix update handler to not delete banner

hariombalhara committed:
code cleanup

sean-brydon committed:
add banner-upload.e2e

sean-brydon committed:
Merge remote-tracking branch 'refs/remotes/origin/fix/bannerurl-being-detatched-from-orgs' into fix/bannerurl-being-detatched-from-orgs

joeauyeung commented on packages/trpc/server/routers/viewer/organizations/update.handler.ts:
```diff
@@ -15,6 +15,38 @@ import { TRPCError } from "@trpc/server";
+  banner: string | null | undefined,
+  teamId: number
+): Promise<string | null | undefined> => {
+  if (banner === undefined) {
+    // Banner not provided, don't update
+    return undefined;
+  }
+
+  if (banner === null) {
+    // Explicitly set to null, remove banner
+    return null;
+  }
```
If we don't pass the bannerUrl but we return `undefined` in this function, won't this overwrite the stored value in the database?

devin-ai-integration[bot] committed:
fix: add slug fallback in banner-upload e2e test

- Use org.slug as fallback when requestedSlug is undefined
- Addresses GitHub comment about guarding against undefined slug
- Prevents test failures when requestedSlug is not set
- Fix ESLint warnings: remove unnecessary escape characters and unused variable

Co-Authored-By: sean@cal.com <Sean@brydon.io>

sean-brydon commented on apps/web/playwright/organization/banner-upload.e2e.ts:
```diff
@@ -0,0 +1,186 @@
+
+    let objectKey: string;
+
+    await test.step("Can upload an initial banner", async () => {
```
Sure

devin-ai-integration[bot] committed:
test: add banner preservation test for profile updates

- Verify that updating organization profile without banner changes preserves existing banner
- Ensures update handler doesn't overwrite banner when undefined is passed
- Addresses user request to test banner preservation during profile saves

Co-Authored-By: sean@cal.com <Sean@brydon.io>

sean-brydon committed:
Merge branch 'main' into fix/bannerurl-being-detatched-from-orgs

sean-brydon committed:
remove banner e2e wip

sean-brydon committed:
Merge branch 'main' into fix/bannerurl-being-detatched-from-orgs
